packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -1301,13 +1303,31 @@ export function returnJsonArray(jsonData: IDataObject | IDataObject[]): INodeExe
 		jsonData = [jsonData];
 	}
 
-	jsonData.forEach((data) => {
+	jsonData.forEach((data: IDataObject) => {
 		returnData.push({ json: data });
 	});
 
 	return returnData;
 }
 
+/**
+ * Takes generic input data and brings it into the new json, pairedItem format n8n uses.
+ * @export
+ * @param {(IPairedItemData)} itemData
+ * @param {(INodeExecutionData[])} inputData
+ * @returns {(INodeExecutionMetaData[])}
+ */
+export function constructExecutionMetaData(
+	itemData: IPairedItemData | IPairedItemData[],
```

Comment:
I suggest we change the argument order and make the second argument (with the itemData) an object called `options`.